node/src/ledger/ledger_service.rs

```diff
@@ -164,6 +161,54 @@ impl LedgerCtx {
             .or_else(|| self.sync.mask(hash))
     }
 
+    fn copy_snarked_ledger_contents(
+        &mut self,
+        origin_snarked_ledger_hash: LedgerHash,
+        target_snarked_ledger_hash: LedgerHash,
+        overwrite: bool,
+    ) -> Result<bool, String> {
+        if !overwrite
+            && self
+                .snarked_ledgers
+                .contains_key(&target_snarked_ledger_hash)
+        {
+            return Ok(false);
+        }
+
+        let origin = self
+            .snarked_ledgers
+            .get_mut(&origin_snarked_ledger_hash)
+            .ok_or(format!(
+                "Tried to copy from non-existing snarked ledger with hash: {}",
+                origin_snarked_ledger_hash.to_string()
+            ))?;
+
+        let target = origin.copy();
+        self.snarked_ledgers
```

Comment:
@binier ufff good catch, when cleaning up some code at some point I got the impression that I was starting to see less reuse than I was expecting, this would explain it.

Reply:
FYI this small change reduced the 3 snarked ledgers sync time from 60s to 20s, so it was indeed skipping all the reuse because of that mistake.